src/reactive-handler.ts

```diff
@@ -178,11 +182,16 @@ export class ReactiveProxyHandler {
         // if the descriptor has a value, as opposed to getter/setter
         // So we can just check if writable is present and then see if
         // value is present. This eliminates getter and setter descriptors
-        if (hasOwnProperty.call(descriptor, 'writable') && !hasOwnProperty.call(descriptor, 'value')) {
-            const originalDescriptor = getOwnPropertyDescriptor(originalTarget, key) as PropertyDescriptor;
+        const originalDescriptor = getOwnPropertyDescriptor(originalTarget, key);
+        if (!isUndefined(originalDescriptor) && hasOwnProperty.call(descriptor, 'writable') && !hasOwnProperty.call(descriptor, 'value')) {
```

Comment:
```suggestion
        if (originalDescriptor && hasOwnProperty.call(descriptor, 'writable') && !hasOwnProperty.call(descriptor, 'value')) {
```

Reply:
all these libraries are using `isUndefined` helper for two reasons:

* it is supposed to be faster because it doesn't have to cast the value.
* It is easier to read and reasoning about what the condition is about.